### ISSUE TITLE
Update Lands boundaries with 20160331 release

### ISSUE THREAD

**Issue by nateirwin** [nationalparkservice/places-data#697]
Released 3/31/2016:

> The "Administrative Boundaries of National Park System Units 3/31/2016 - National Geospatial Data Asset (NGDA) NPS National Parks Dataset" GIS data located on the IRMA Portal  has been updated.  To access this data go to the IRMA home page (https://irma.nps.gov/App/Portal/Home) and click on "Data Store" and then click on "Park Boundaries" under the "Frequently Requested" header.

New units in this release:
- Cape Cod National Seashore (CACO)  
- Castle Mountains National Monument (IRMA reference # 2229071) (CAMO)
- Congaree National Park (CONG)
- Frederick Law Olmsted National Historic Site (FRLA)
- George Washington Birthplace National Monument (GEWA)
- Manhattan Project National Historical Park (MAPR)
- National Park of American Samoa (NPSA)
- Tuskegee Institute National Historic Site (TUIN)
- Waco Mammoth National Monument (WACO)

Updated units in this release:
- Glen Canyon National Recreation Area (GLCA)
- Hamilton Grange National Memorial (HAGR)
- Hampton National Historic Site (HAMP)
- Indiana Dunes National Lakeshore (INDU)
- Lincoln Boyhood National Memorial (LIBO)
- Lower Saint Croix National Scenic Riverway (LOSA)
- Mojave National Preserve (MOJA)
- Mount Rainier National Park (MORA)
- San Antonio Missions National Historical Park (SAAN)
- Saratoga National Historical Park (SARA)
- Tallgrass Prairie National Preserve (TAPR)
- Wilson's Creek National Battlefield (WICR)
- Yosemite National Park (YOSE)

**Comment by chadlawlis** [comment]
I was just writing up the same issue :+1: 

New table `lands_20160331` has been added to `places_boundaries`.

**Comment by chadlawlis** [comment]
Of the new units in this release, we are only missing a record for CAMO, while FRLA is the only existing record missing a boundary. The current data sources are as follows:

| unit_id | name | unit_code | unit_region | data_source | geom_poly | poly_last_updated |
| --- | --- | --- | --- | --- | --- | --- |
| 53 | Cape Cod National Seashore | caco | ner | WSD_Parks | y |  |
| 90 | Congaree National Park | cong | ser | Land Resources Division | y | 2016-02-10 12:44:36.970809-07 |
| 153 | Frederick Law Olmsted National Historic Site | frla | ner | Land Resources Division |  | 2016-02-29 10:59:14.368277-07 |
| 162 | George Washington Birthplace National Monument | gewa | ner | WSD_Parks | y |  |
| 452 | Manhattan Project National Historical Park | mapr |  | mapr | y | 2016-02-11 13:54:20.292837-07 |
| 298 | National Park of American Samoa | npsa | pwr | WSD_Parks | y |  |
| 386 | Tuskegee Institute National Historic Site | tuin | ser | Land Resources Division | y | 2016-02-10 12:44:36.970809-07 |
| 429 | Waco Mammoth National Monument | waco | imr | imr | y | 2015-12-28 13:04:01.331368-07 |

All of these will be updated with this release.

**Comment by chadlawlis** [comment]
CAMO process available here: #546.

All other new units updated via:

``` sql
BEGIN;
-- CACO
UPDATE
    parks_poly
SET
    geom_poly = (SELECT wkb_geometry FROM lands_20160331 WHERE unit_code = 'caco')
WHERE
    unit_code = 'caco';

UPDATE 
    parks
SET
    geom_point = (SELECT ST_Centroid(geom_poly) FROM parks_poly WHERE unit_code = 'caco'),
    unit_area = (SELECT ST_Area(geom_poly) FROM parks_poly WHERE unit_code = 'caco'),
    data_source = 'Land Resources Division'
WHERE
    unit_code = 'caco';

-- CONG
UPDATE
    parks_poly
SET
    geom_poly = (SELECT wkb_geometry FROM lands_20160331 WHERE unit_code = 'cong')
WHERE
    unit_code = 'cong';

UPDATE 
    parks
SET
    geom_point = (SELECT ST_Centroid(geom_poly) FROM parks_poly WHERE unit_code = 'cong'),
    unit_area = (SELECT ST_Area(geom_poly) FROM parks_poly WHERE unit_code = 'cong'),
    data_source = 'Land Resources Division'
WHERE
    unit_code = 'cong';

-- FRLA
UPDATE
    parks_poly
SET
    geom_poly = (SELECT wkb_geometry FROM lands_20160331 WHERE unit_code = 'frla')
WHERE
    unit_code = 'frla';

UPDATE 
    parks
SET
    geom_point = (SELECT ST_Centroid(geom_poly) FROM parks_poly WHERE unit_code = 'frla'),
    unit_area = (SELECT ST_Area(geom_poly) FROM parks_poly WHERE unit_code = 'frla'),
    data_source = 'Land Resources Division'
WHERE
    unit_code = 'frla';

-- GEWA
UPDATE
    parks_poly
SET
    geom_poly = (SELECT wkb_geometry FROM lands_20160331 WHERE unit_code = 'gewa')
WHERE
    unit_code = 'gewa';

UPDATE 
    parks
SET
    geom_point = (SELECT ST_Centroid(geom_poly) FROM parks_poly WHERE unit_code = 'gewa'),
    unit_area = (SELECT ST_Area(geom_poly) FROM parks_poly WHERE unit_code = 'gewa'),
    data_source = 'Land Resources Division'
WHERE
    unit_code = 'gewa';

-- MAPR
UPDATE
    parks_poly
SET
    geom_poly = (SELECT wkb_geometry FROM lands_20160331 WHERE unit_code = 'mapr')
WHERE
    unit_code = 'mapr';

UPDATE 
    parks
SET
    geom_point = (SELECT ST_Centroid(geom_poly) FROM parks_poly WHERE unit_code = 'mapr'),
    unit_area = (SELECT ST_Area(geom_poly) FROM parks_poly WHERE unit_code = 'mapr'),
    data_source = 'Land Resources Division'
WHERE
    unit_code = 'mapr';

-- NPSA
UPDATE
    parks_poly
SET
    geom_poly = (SELECT wkb_geometry FROM lands_20160331 WHERE unit_code = 'npsa')
WHERE
    unit_code = 'npsa';

UPDATE 
    parks
SET
    geom_point = (SELECT ST_Centroid(geom_poly) FROM parks_poly WHERE unit_code = 'npsa'),
    unit_area = (SELECT ST_Area(geom_poly) FROM parks_poly WHERE unit_code = 'npsa'),
    data_source = 'Land Resources Division'
WHERE
    unit_code = 'npsa';

-- TUIN
UPDATE
    parks_poly
SET
    geom_poly = (SELECT wkb_geometry FROM lands_20160331 WHERE unit_code = 'tuin')
WHERE
    unit_code = 'tuin';

UPDATE 
    parks
SET
    geom_point = (SELECT ST_Centroid(geom_poly) FROM parks_poly WHERE unit_code = 'tuin'),
    unit_area = (SELECT ST_Area(geom_poly) FROM parks_poly WHERE unit_code = 'tuin'),
    data_source = 'Land Resources Division'
WHERE
    unit_code = 'tuin';

-- WACO
UPDATE
    parks_poly
SET
    geom_poly = (SELECT wkb_geometry FROM lands_20160331 WHERE unit_code = 'waco')
WHERE
    unit_code = 'waco';

UPDATE 
    parks
SET
    geom_point = (SELECT ST_Centroid(geom_poly) FROM parks_poly WHERE unit_code = 'waco'),
    unit_area = (SELECT ST_Area(geom_poly) FROM parks_poly WHERE unit_code = 'waco'),
    data_source = 'Land Resources Division'
WHERE
    unit_code = 'waco';
COMMIT;
```

**Comment by chadlawlis** [comment]
We have a record for all updated units in this release, each with a boundary. Current status as follows:

| unit_id | name | unit_code | unit_region | data_source | geom_poly | poly_last_updated |
| --- | --- | --- | --- | --- | --- | --- |
| 166 | Glen Canyon National Recreation Area | glca | imr | Land Resources Division | y | 2016-03-28 15:37:04.112212-06 |
| 188 | Hamilton Grange National Memorial | hagr | ner | Land Resources Division | y |  |
| 191 | Hampton National Historic Site | hamp | ner | Land Resources Division | y |  |
| 206 | Indiana Dunes National Lakeshore | indu | mwr | WSD_Parks | y |  |
| 245 | Lincoln Boyhood National Memorial | libo | mwr | WSD_Parks | y |  |
| 250 | Lower Saint Croix National Scenic Riverway | losa | mwr | Land Resources Division | y |  |
| 275 | Mojave National Preserve | moja | pwr | WSD_Parks | y |  |
| 277 | Mount Rainier National Park | mora | pwr | MORA | y |  |
| 345 | San Antonio Missions National Historical Park | saan | imr | Land Resources Division | y | 2016-03-28 15:37:04.112212-06 |
| 360 | Saratoga National Historical Park | sara | ner | WSD_Parks | y |  |
| 374 | Tallgrass Prairie National Preserve | tapr | mwr | WSD_Parks | y |  |
| 412 | Wilson's Creek National Battlefield | wicr | mwr | WSD_Parks | y |  |
| 421 | Yosemite National Park | yose | pwr | WSD_Parks | y |  |

**Comment by chadlawlis** [comment]
All of the above (updated) parks currently pulling from Lands were updated via:

``` sql
BEGIN;
UPDATE
    parks_poly
SET
    geom_poly = (SELECT wkb_geometry FROM lands_20160331 WHERE unit_code = 'glca')
WHERE
    unit_code = 'glca';

UPDATE 
    parks
SET
    unit_area = (SELECT ST_Area(geom_poly) FROM parks_poly WHERE unit_code = 'glca')
WHERE
    unit_code = 'glca';


UPDATE
    parks_poly
SET
    geom_poly = (SELECT wkb_geometry FROM lands_20160331 WHERE unit_code = 'hagr')
WHERE
    unit_code = 'hagr';

UPDATE 
    parks
SET
    unit_area = (SELECT ST_Area(geom_poly) FROM parks_poly WHERE unit_code = 'hagr')
WHERE
    unit_code = 'hagr';


UPDATE
    parks_poly
SET
    geom_poly = (SELECT wkb_geometry FROM lands_20160331 WHERE unit_code = 'hamp')
WHERE
    unit_code = 'hamp';

UPDATE 
    parks
SET
    unit_area = (SELECT ST_Area(geom_poly) FROM parks_poly WHERE unit_code = 'hamp')
WHERE
    unit_code = 'hamp';


UPDATE
    parks_poly
SET
    geom_poly = (SELECT wkb_geometry FROM lands_20160331 WHERE unit_code = 'losa')
WHERE
    unit_code = 'losa';

UPDATE 
    parks
SET
    unit_area = (SELECT ST_Area(geom_poly) FROM parks_poly WHERE unit_code = 'losa')
WHERE
    unit_code = 'losa';


UPDATE
    parks_poly
SET
    geom_poly = (SELECT wkb_geometry FROM lands_20160331 WHERE unit_code = 'saan')
WHERE
    unit_code = 'saan';

UPDATE 
    parks
SET
    unit_area = (SELECT ST_Area(geom_poly) FROM parks_poly WHERE unit_code = 'saan')
WHERE
    unit_code = 'saan';
COMMIT;
```

**Comment by chadlawlis** [comment]
Waiting on MORA via https://github.com/nationalparkservice/places-data/issues/258#issuecomment-207621626

**Comment by chadlawlis** [comment]
MOJA transitioned to Lands via #700.

**Comment by chadlawlis** [comment]
WSD_Parks and Lands are borderline identical for LIBO, so transitioning to Lands:

![libo](https://cloud.githubusercontent.com/assets/1238636/14398682/484fc984-fda3-11e5-9cb6-ebef6062d6ad.png)

Done via:

``` sql
BEGIN;
UPDATE
    parks_poly
SET
    geom_poly = (SELECT wkb_geometry FROM lands_20160331 WHERE unit_code = 'libo')
WHERE
    unit_code = 'libo';

UPDATE 
    parks
SET
    unit_area = (SELECT ST_Area(geom_poly) FROM parks_poly WHERE unit_code = 'libo'),
    data_source = 'Land Resources Division'
WHERE
    unit_code = 'libo';
COMMIT;
```

**Comment by chadlawlis** [comment]
WSD_Parks and Lands are borderline identical for TAPR, so transitioning to Lands:

![tapr](https://cloud.githubusercontent.com/assets/1238636/14398755/d9f46a20-fda3-11e5-8dce-b662101c4d58.png)

Done via:

``` sql
BEGIN;
UPDATE
    parks_poly
SET
    geom_poly = (SELECT wkb_geometry FROM lands_20160331 WHERE unit_code = 'tapr')
WHERE
    unit_code = 'tapr';

UPDATE 
    parks
SET
    unit_area = (SELECT ST_Area(geom_poly) FROM parks_poly WHERE unit_code = 'tapr'),
    data_source = 'Land Resources Division'
WHERE
    unit_code = 'tapr';
COMMIT;
```

**Comment by chadlawlis** [comment]
WSD_Parks and Lands are identical for WICR, so transitioning to Lands:

![wicr](https://cloud.githubusercontent.com/assets/1238636/14398800/39a5da94-fda4-11e5-8188-c5548b95e77a.png)

Done via:

``` sql
BEGIN;
UPDATE
    parks_poly
SET
    geom_poly = (SELECT wkb_geometry FROM lands_20160331 WHERE unit_code = 'wicr')
WHERE
    unit_code = 'wicr';

UPDATE 
    parks
SET
    unit_area = (SELECT ST_Area(geom_poly) FROM parks_poly WHERE unit_code = 'wicr'),
    data_source = 'Land Resources Division'
WHERE
    unit_code = 'wicr';
COMMIT;
```

**Comment by chadlawlis** [comment]
WSD_Parks and Lands are borderline identical for YOSE, so transitioning to Lands:

![yose](https://cloud.githubusercontent.com/assets/1238636/14398831/7ad37b7a-fda4-11e5-84d5-e63e14a88c93.png)

Done via:

``` sql
BEGIN;
UPDATE
    parks_poly
SET
    geom_poly = (SELECT wkb_geometry FROM lands_20160331 WHERE unit_code = 'yose')
WHERE
    unit_code = 'yose';

UPDATE 
    parks
SET
    unit_area = (SELECT ST_Area(geom_poly) FROM parks_poly WHERE unit_code = 'yose'),
    data_source = 'Land Resources Division'
WHERE
    unit_code = 'yose';
COMMIT;
```

**Comment by chadlawlis** [comment]
The WSD_Parks boundaries for INDU and SARA are drastically different than Lands. INDU has already been separated into its own issue #433, to be addressed there. SARA to be addressed via #706.

MORA is the only other that was not immediately updated, the link to which is found above. This is now waiting on the tiles to be regenerated, once complete I will close this out.

**Comment by chadlawlis** [comment]
Tiles have been regenerated, these are now live.
